internal/kv/tikv/raw_tikv.go

```diff
@@ -0,0 +1,347 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tikv
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	tikv "github.com/tikv/client-go/v2/kv"
+	"github.com/tikv/client-go/v2/rawkv"
+	"go.uber.org/zap"
+
+	"github.com/milvus-io/milvus/internal/kv"
+	"github.com/milvus-io/milvus/pkg/common"
+	"github.com/milvus-io/milvus/pkg/log"
+	"github.com/milvus-io/milvus/pkg/metrics"
+	"github.com/milvus-io/milvus/pkg/util/timerecord"
+)
+
+const (
+	MaxScanLimit = 10240
+)
+
+// implementation assertion
+var _ kv.MetaKv = (*rawTiKV)(nil)
+
+// rawTiKV implements MetaKv (non-TxnKV part) interface. It supports processing multiple kvs in a batch(not transaction) manner.
+type rawTiKV struct {
+	client   *rawkv.Client
+	rootPath string
+}
+
+// NewRawTiKV creates a new rawTiKV client.
+func NewRawTiKV(rawkv *rawkv.Client, rootPath string) *rawTiKV {
+	kv := &rawTiKV{
+		client:   rawkv,
+		rootPath: rootPath,
+	}
+	return kv
+}
+
+// Close closes the connection to TiKV.
+func (kv *rawTiKV) Close() {
+	log.Debug("rawTiKV closed", zap.String("path", kv.rootPath))
+}
+
+// GetPath returns the path of the key/prefix.
+func (kv *rawTiKV) GetPath(key string) string {
+	return path.Join(kv.rootPath, key)
+}
+
+// Has returns if a key exists.
+func (kv *rawTiKV) Has(key string) (bool, error) {
+	start := time.Now()
+	key = path.Join(kv.rootPath, key)
+	ctx, cancel := context.WithTimeout(context.Background(), RequestTimeout)
```

Comment:
The ctx should in parameters 🤔, we can refactor it in another PR

Reply:
The impact of the changes is quite significant.

Reply:
It's actually not used.
Deleted the entire file.